### PR TITLE
Issue 5

### PR DESCRIPTION
**Pull request by h714714**

### CODE REVIEW

**Comment by h714714** [comment]
[Issue 5: Image Tags](https://trello.com/c/s0xzVqyG/199-issue-5-image-tags)

File: app/models/image.rb [hunk 1]
```diff
@@ -4,4 +4,5 @@ class Image < ApplicationRecord
   validates_format_of :image_url,
                       with: %r{^(http|https)://[a-z0-9]+([\-\.]{1}[a-z0-9]+)*\.[a-z]{2,5}(:[0-9]{1,5})?(/.*)?$}ix, # rubocop:disable Metrics/LineLength
                       multiline: true
+  acts_as_taggable
```
Comment (eldonchen0501) [inline app/models/image.rb]: This changes associates the tags to the image model. Instead of being put in this commit, I think we should have an individual commit for it.

File: test/controllers/images_controller_test.rb [hunk 2]
```diff
@@ -52,4 +52,23 @@ def test_homepage
...
+
+      post images_path, params: { 'image' => { 'image_url' => 'http://efg.com', 'tag_list' => 'e, f, g' } }
+      assert_equal Image.last.tag_list, %w[e f g]
+
+      post images_path, params: { 'image' => { 'image_url' => 'http://empty.com' } }
+      assert_equal Image.last.tag_list, []
+    end
+  end
+
+  def test_create_invalid_url_tag
```
Comment (eldonchen0501) [inline test/controllers/images_controller_test.rb]: this is a redundant test and the name of the test doesn't match being tested here

File: test/controllers/images_controller_test.rb [hunk 3]
```diff
@@ -52,4 +52,23 @@ def test_homepage
...
+  def test_create_valid_url_tag
+    assert_difference 'Image.count', 3 do
+      post images_path, params: { 'image' => { 'image_url' => 'http://a.com', 'tag_list' => 'a' } }
+      assert_equal Image.last.tag_list, %w[a]
+
+      post images_path, params: { 'image' => { 'image_url' => 'http://efg.com', 'tag_list' => 'e, f, g' } }
+      assert_equal Image.last.tag_list, %w[e f g]
+
+      post images_path, params: { 'image' => { 'image_url' => 'http://empty.com' } }
+      assert_equal Image.last.tag_list, []
```
Comment (eldonchen0501) [inline test/controllers/images_controller_test.rb]: we should be verifying that there are threes tags instead of zero.

Comment (h714714) [reply test/controllers/images_controller_test.rb]: But this last image is created without tag, so there should be no tag.

Comment (eldonchen0501) [reply test/controllers/images_controller_test.rb]: 👍

File: test/controllers/images_controller_test.rb (hunk 2 above)
Comment (h714714) [reply test/controllers/images_controller_test.rb]: removed

File: test/controllers/images_controller_test.rb (hunk 3 above)
Comment (h714714) [reply test/controllers/images_controller_test.rb]: put assertion together

File: app/models/image.rb (hunk 1 above)
Comment (h714714) [reply app/models/image.rb]: add new commit and rebase the order of it

File: test/controllers/images_controller_test.rb [hunk 4]
```diff
@@ -56,19 +56,11 @@ def test_homepage
...
-
       post images_path, params: { 'image' => { 'image_url' => 'http://empty.com' } }
-      assert_equal Image.last.tag_list, []
-    end
-  end
-
-  def test_create_invalid_url_tag
-    assert_no_difference 'Image.count' do
-      post images_path, params: { 'image' => { 'image_url' => '123456', 'tag_list' => 'a' } }
+      assert_equal Image.last(3)[0].tag_list, %w[a]
```
Comment (eldonchen0501) [inline test/controllers/images_controller_test.rb]: this change should belong to another commit

File: app/controllers/images_controller.rb [hunk 5]
```diff
@@ -17,6 +17,7 @@ def create
   def show
     image = Image.find(params[:id])
     @url = image.image_url
+    @tags = image.tag_list
```
Comment (eldonchen0501) [inline app/controllers/images_controller.rb]: this should belong to the commit for show action
